pkg/endpoint/policy.go

```diff
@@ -210,7 +210,17 @@ func (e *Endpoint) determineAllowLocalhost(desiredPolicyKeys map[policymap.Polic
 			Identity:         identityPkg.ReservedIdentityHost.Uint32(),
```

Comment:
Hmm, I don't follow the logic above with `|| (e.DesiredL4Policy != nil && e.DesiredL4Policy.HasRedirect())` but would we allow all connections to the host if only `e.DesiredL4Policy.HasRedirect()` was true? I'm guessing those are added to allow communicating with the L7 proxy and that should remain less permissive than the allow-all case. I don't think I understand this, though, since the logic was already broken in that case.

Reply:
I think the `HasRedirect()` can be closed as the datapath relies on skb marks to identify proxy packets. I did not want to change it in this PR though.